app/grandchallenge/evaluation/views/__init__.py

```diff
@@ -1384,6 +1385,46 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class AlgorithmInterfacesForPhaseCopy(
+    ConfigureAlgorithmPhasesPermissionMixin,
+    AlgorithmInterfaceForPhaseMixin,
+    SuccessMessageMixin,
+    FormView,
+):
```

Comment:
I think we might need to look this down further to make sure that we're not accidentally updating locked phases (i.e. those that are in a parent-child relationship). We also lock the other interface views for those phases. There is a property on the phase model you can use for that `algorithm_interfaces_locked`, see also `LockParentAndChildPhasesMixin`

Reply:
The subsetting would best be done in the form, not here, of course.

Reply:
Good idea. I've made the changes.